rbac/rbac/middleware.py

```diff
@@ -386,18 +386,17 @@ def process_response(self, request, response):  # pylint: disable=no-self-use
             request (object): The request object
             response (object): The response object
         """
-        is_internal = False
-        if any([request.path.startswith(prefix) for prefix in settings.INTERNAL_API_PATH_PREFIXES]):
-            # This request is for a private API endpoint
-            is_internal = True
-            IdentityHeaderMiddleware.log_request(request, response, is_internal)
-            return response
-
-        behalf = "principal"
+        is_internal = any([request.path.startswith(prefix) for prefix in settings.INTERNAL_API_PATH_PREFIXES])
         is_system = False
 
-        if is_system:
-            behalf = "system"
+        if hasattr(request, "user") and request.user:
+            username = request.user.username
+            if username:
+                is_system = request.user.system
+            else:
+                is_system = False
```

Comment:
LGTM, just nitpick ... I think the else is not needed here, because the False value is set few lines above